delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/GpuDeltaParquetFileFormatUtils.scala

```diff
@@ -98,28 +102,30 @@ object GpuDeltaParquetFileFormatUtils {
     closeOnExcept(rowIdxCol) { rowIdxCol =>
 
       val delVecCol = delVec.map { delVec =>
-        withResource(Scalar.fromBool(false)) { s =>
-          withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
-            var table = new Table(c)
-            val posIter = new RoaringBitmapIterator(
-              delVec.getLongIteratorFrom(rowIdxStart),
-              rowIdxStart,
-              rowIdxStart + batch.numRows(),
-            ).grouped(Math.min(maxBatchSize, batch.numRows()))
-
-            for (posChunk <- posIter) {
-              withResource(CudfColumnVector.fromLongs(posChunk: _*)) { poses =>
-                withResource(Scalar.fromBool(true)) { s =>
-                  table = withResource(table) { _ =>
-                    Table.scatter(Array(s), poses, table)
+        delVectorScatterTimeMetric.ns {
+          withResource(Scalar.fromBool(false)) { s =>
+            withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
+              var table = new Table(c)
+              val posIter = new RoaringBitmapIterator(
+                delVec.getLongIteratorFrom(rowIdxStart),
+                rowIdxStart,
+                rowIdxStart + batch.numRows(),
+              ).grouped(Math.min(maxBatchSize, batch.numRows()))
+
+              for (posChunk <- posIter) {
+                withResource(CudfColumnVector.fromLongs(posChunk: _*)) { poses =>
+                  withResource(Scalar.fromBool(true)) { s =>
+                    table = withResource(table) { _ =>
+                      Table.scatter(Array(s), poses, table)
+                    }
```

Comment:
`table` will leak if creating either `poses` or `s` throws.
`s` is constant for the loop, so creating it within the loop adds an unnecessary GPU memcpy and synchronization each iteration after the first.
```suggestion
              val s = closeOnExcept(table) { _ =>
                Scalar.fromBool(true)
              }
              withResource(s) { _ =>
                for (posChunk <- posIter) {
                  table = withResource(table) { _ =>
                    withResource(CudfColumnVector.fromLongs(posChunk: _*)) { poses =>
                      Table.scatter(Array(s), poses, table)
                    }
```